servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererBuilder.java

```diff
@@ -182,19 +182,46 @@ DnsServiceDiscovererBuilder dnsServerAddressStreamProvider(
 
     /**
      * Set the number of dots which must appear in a name before an initial absolute query is made.
+     * <p>
+     * If not set, the default value is read from {@code ndots} option of {@code /etc/resolv.conf}).
      *
      * @param ndots the ndots value.
      * @return {@code this}.
      */
     DnsServiceDiscovererBuilder ndots(int ndots);
 
     /**
-     * Sets the timeout of each DNS query performed by this service discoverer.
+     * Sets the timeout of each DNS query performed by this service discoverer as part of a resolution request.
+     * <p>
+     * Zero ({@code 0}) disables the timeout. If not set, the default value is read from {@code timeout} option of
+     * {@code /etc/resolv.conf}). Similar to linux systems, this value may be silently capped.
      *
      * @param queryTimeout the query timeout value
-     * @return {@code this}.
+     * @return {@code this}
+     * @see #resolutionTimeout(Duration)
+     */
+    DnsServiceDiscovererBuilder queryTimeout(@Nullable Duration queryTimeout);
+
+    /**
+     * Sets the total timeout of each DNS resolution performed by this service discoverer.
+     * <p>
+     * Each resolution may execute one or more DNS queries, like following multiple CNAME(s) or trying different search
+     * domains. This is the total timeout for all intermediate queries involved in a single resolution request. Note,
+     * that <a href="https://tools.ietf.org/html/rfc2782">SRV</a> resolutions may generate independent resolutions for
```

Comment:
this is a slightly different conversations, but I wonder if this is the right behavior. Even if technically it's two different resolutions, from a user perspective I still want to apply a "total" timeout, right? Since in the end they are just individual queries against a nameserver where we need a total timeout to cap?

Reply:
agreed, I don't like this fact either but it's what it's. the way SRV resolutions are implemented right now makes it too complicated to apply a timeout at SRV+all_A_resolutions level. requires a big refactoring. 
taking into account, they are less frequent and used only for `BACKGROUND` discovery strategy, the need for a proper total timeout is less.